component/client/chat/ChatHistory.tsx

```diff
@@ -3,111 +3,62 @@ import {ListGroup} from "react-bootstrap";
 import * as ReactDOM from "react-dom";
 
 import IData from "../service/IData";
-import RoomService from "../service/RoomService";
 import Message from "./Message";
 
 interface IHistoryState {
-  messages: JSX.Element[];
 }
 
 interface IHistoryProps {
-  username: string;
-  url: string;
-  pushMessage: string;
+  messages: IData[];
 }
 
 class ChatHistory extends React.Component<IHistoryProps, IHistoryState> {
-  private roomService: RoomService;
   private historyEnd: HTMLElement;
   private shouldScroll: boolean;
   private lastPosition: number;
 
   constructor(props: any) {
     super(props);
-
-    this.state = {
-      messages: []
-    };
     this.shouldScroll = true;
-
-    this.instantiateRoomService = this.instantiateRoomService.bind(this);
-    this.createMessage = this.createMessage.bind(this);
     this.handleScroll = this.handleScroll.bind(this);
     this.setHistoryEnd = this.setHistoryEnd.bind(this);
   }
 
   public render(): JSX.Element {
+    const listGroupStyle = {
+      marginBottom: "3px",
+      marginTop: "3px"
+    };
+    console.log(this.props.messages);
     return (
       <div style={{overflow: "auto", height: "268px", minHeight: "268px", maxHeight: "268px"}} onScroll={this.handleScroll}>
-        {this.state.messages}
+        {
+          this.props.messages.map((data, index)=>
+            <ListGroup key={index + data.userFrom + data.message} style={listGroupStyle}>
+              <Message username={data.userFrom} message={data.message} />
+            </ListGroup>
+          )
+        }
```

Comment:
This is redundant. Every time we receive one message, all messages are being re-constructed into a Message component. 

Before, we had a state array of messages that maintain old messages component because they shouldn't be changing and pushed (concatted) the new message data on.